src/c-writer.cc

```diff
@@ -1853,20 +1853,37 @@ void CWriter::WriteDataInstances() {
   }
 }
 
+// Determines whether or not the selected data range contains only zero bytes.
+static bool is_all_zero(const DataSegment* segment,
+                        size_t start_index,
+                        size_t end_index) {
+  while (start_index < end_index) {
+    if (segment->data[start_index] != 0) {
+      return false;
+    }
+    ++start_index;
+  }
+  return true;
+}
```

Comment:
Hmm, it never occurred to me that there was not C stdlib function for this.. kind of the inverse `memset`.

Reply:
I bet there is some function that can do that somewhere, but I don't know where that would be.